src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -35,21 +74,46 @@ protected function processValue($value, bool $isRoot = false)
             || !$value->isAutoconfigured()
             || $value->isAbstract()
             || $value->hasTag('container.ignore_attributes')
-            || !($reflector = $this->container->getReflectionClass($value->getClass(), false))
+            || !($classReflector = $this->container->getReflectionClass($value->getClass(), false))
         ) {
             return parent::processValue($value, $isRoot);
         }
 
-        $autoconfiguredAttributes = $this->container->getAutoconfiguredAttributes();
         $instanceof = $value->getInstanceofConditionals();
-        $conditionals = $instanceof[$reflector->getName()] ?? new ChildDefinition('');
-        foreach ($reflector->getAttributes() as $attribute) {
-            if ($configurator = $autoconfiguredAttributes[$attribute->getName()] ?? null) {
-                $configurator($conditionals, $attribute->newInstance(), $reflector);
+        $conditionals = $instanceof[$classReflector->getName()] ?? new ChildDefinition('');
+
+        if (count($this->classAttributes) > 0) {
+            foreach ($classReflector->getAttributes() as $attribute) {
+                if ($configurator = $this->classAttributes[$attribute->getName()] ?? null) {
+                    $configurator($conditionals, $attribute->newInstance(), $classReflector);
+                }
+            }
+        }
+
+        if (count($this->methodAttributes) > 0) {
+            foreach ($classReflector->getMethods(\ReflectionMethod::IS_PUBLIC & ~\ReflectionMethod::IS_STATIC) as $methodReflector) {
+                foreach ($methodReflector->getAttributes() as $attribute) {
+                    if ($configurator = $this->methodAttributes[$attribute->getName()] ?? null) {
+                        $configurator($conditionals, $attribute->newInstance(), $methodReflector);
+                    }
+                }
             }
         }
-        if (!isset($instanceof[$reflector->getName()]) && new ChildDefinition('') != $conditionals) {
-            $instanceof[$reflector->getName()] = $conditionals;
+
+        if (count($this->propertyAttributes) > 0) {
+            foreach ($classReflector->getProperties(~\ReflectionProperty::IS_STATIC) as $propertyReflector) {
```

Comment:
`ReflectionProperty::IS_PUBLIC`, and the check for non-static props should be done by calling the method